prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -417,6 +423,10 @@ class DetailedSubmissionHistory(
                 }
             }
         }
+
+        descendant.reports?.forEach {
```

Comment:
NIT: Why do you need a separate loop for this, why can we not increment the counter inside the same block where we do destinations.add?